### ISSUE TITLE
[Bug] Deadlock when mixing transactions and normal commands in 10.0.1

### ISSUE THREAD

**Issue by ItsEthra** [aembke/fred.rs#320]
Fred version - 10.0.1
Redis version - 7.2.5
Platform - linux
Deployment type - centralized

**Describe the bug**
In multhreaded runtime it's easy to hang when mixing transactions and normal commands. This issue only started to appear in 10.0.0

**To Reproduce**
```toml
[dependencies]
fred = { version = "10.0.1", features = ["debug-ids"] }
tokio = { version = "1.42.0", features = ["rt", "rt-multi-thread", "macros", "net"] }
tracing-subscriber = "0.3.19"
```

```rust
use fred::{
    prelude::{Client, ClientLike, KeysInterface, TransactionInterface},
    types::Builder,
};
use tokio::task::JoinSet;

async fn task(client: Client) {
    let tx = client.multi();
    () = tx.incr("foo").await.unwrap();
    () = tx.exec(true).await.unwrap();

    // hangs here i think
    () = client.get("bar").await.unwrap();
}

// Doesn't happen on single threaded runtime
// #[tokio::main(flavor = "current_thread")]

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt::init();

    let client = Builder::default_centralized().build().unwrap();
    client.init().await.unwrap();

    // increasing the number increases the chances of a deadlock
    let n = 200;

    let mut set = JoinSet::new();
    for _ in 0..n {
        set.spawn(task(client.clone()));
    }

    set.join_all().await;
    eprintln!("End");
}
```

Observe that when `n` is small, program finishes successfully, but when `n` is bigger then it hangs.

**Logs**
(If possible set `RUST_LOG=fred=trace` and run with `--features debug-ids`)
The log is 13k lines long since I had to set `n = 800` to get it to hang, it's harder for the program to hang if it's logging stuff.
[log.txt](https://github.com/user-attachments/files/18222058/log.txt)

**Comment by aembke** [comment]
Thanks @ItsEthra, this should be fixed in 10.0.2 (https://github.com/aembke/fred.rs/pull/321)

**Comment by ItsEthra** [comment]
Much thanks! It seems to be working now.
